src/mscorlib/shared/System/Threading/ExecutionContext.cs

```diff
@@ -264,12 +358,14 @@ internal static void SetLocalValue(IAsyncLocal local, object newValue, bool need
                 }
             }
 
-            Thread.CurrentThread.ExecutionContext =
-                new ExecutionContext(newValues, newChangeNotifications, current.m_isFlowSuppressed);
+            Thread.CurrentThread.ExecutionContext = 
+                (!isFlowSuppressed && newValues.GetType() == typeof(AsyncLocalValueMap.EmptyAsyncLocalValueMap)) ?
```

Comment:
In what situation could newValues be empty here?

Reply:
Ah, if null was set as the value for an already empty map I guess?

Reply:
If the value in `OneElementAsyncLocalValueMap` is set to null